### ISSUE TITLE
Issue with tutorials/metrics-export-with-mql

### ISSUE THREAD

**Issue by ljalvrdz** [GoogleCloudPlatform/community#1619]
Hello I just deployed the metrics export with MQL and it just worked once, I only see two records and the Cron has been running for hours. I see this error within Cloud Function logging:
```
KeyError: 'timeSeriesDescriptor'
at save_to_bq (/workspace/main.py:226)
at export_metric_data (/workspace/main.py:246)
at view_func (/layers/google.python.pip/pip/lib/python3.8/site-packages/functions_framework/__init__.py:102)
at dispatch_request (/layers/google.python.pip/pip/lib/python3.8/site-packages/flask/app.py:1936)
at full_dispatch_request (/layers/google.python.pip/pip/lib/python3.8/site-packages/flask/app.py:1950)
at reraise (/layers/google.python.pip/pip/lib/python3.8/site-packages/flask/_compat.py:39)
at handle_user_exception (/layers/google.python.pip/pip/lib/python3.8/site-packages/flask/app.py:1821)
at full_dispatch_request (/layers/google.python.pip/pip/lib/python3.8/site-packages/flask/app.py:1952)
at wsgi_app (/layers/google.python.pip/pip/lib/python3.8/site-packages/flask/app.py:2447)
```
Am I missing something? Any advice will be appreciated.

Regards

**Comment by ljalvrdz** [comment]
Making some more testing I found something interesting, I got this error while printing the result of the request:
```
{'error': {'code': 401, 'message': 'Request had invalid authentication credentials. Expected OAuth 2 access token, login cookie or other valid authentication credential. See https://developers.google.com/identity/sign-in/web/devconsole-project.', 'status': 'UNAUTHENTICATED'}}
```
My testing results match with this as the metric worked for about 30 minutes, that is also the time the token is good for. After some more testing I notice that the variable `token` never changes, it got assigned the first requested token and then it keeps it, it's never renewed so I made a simple change to the entry point to workaround:

```
def export_metric_data(event, context):
    global token
    token = None
    """Background Cloud Function to be triggered by Pub/Sub.
    Args:
         event (dict):  The dictionary with data specific to this type of
         event. The `data` field contains the PubsubMessage message. The
         `attributes` field will contain custom attributes if there are any.
         context (google.cloud.functions.Context): The Cloud Functions event
         metadata. The `event_id` field contains the Pub/Sub message ID. The
         `timestamp` field contains the publish time.
    """
    print("""This Function was triggered by messageId {} published at {}
    """.format(context.event_id, context.timestamp))
    
    token = get_access_token_from_meta_data()
    save_to_bq(token)
```
On the other hand, is this an expected behaivior or am I doing something wrong?

Hope this helps someone else or to improve the capabilities of this useful code.

Thank you!

**Comment by omer-niv** [comment]
same problem, stops working after 30min and I get this:
export_metric_data save_to_bq(token) File "/workspace/main.py", line 226, in save_to_bq if result["timeSeriesDescriptor"]: KeyError: 'timeSeriesDescriptor'

**Comment by ljalvrdz** [comment]
Hi @omer-niv,

If you are still having this problem take a look to my workaround in [comment #2](https://github.com/GoogleCloudPlatform/community/issues/1619#issuecomment-777894584) it works and you should be able to make it work.

Let me know if you have problems.

**Comment by omer-niv** [comment]
Hi @ljalvrdz , I'm afraid it didn't help. It's something with the result trying to write back to BQ.

**Comment by ToddKopriva** [comment]
@xiangshen-dk , can you take a look at this issue about a document that you contributed?

**Comment by ljalvrdz** [comment]
> Hi @ljalvrdz , I'm afraid it didn't help. It's something with the result trying to write back to BQ.

Print _result_ form the save_to_bq function and take a look in the logging, you will get a detailed error. The error you're currently seeing is just the _if_ condition failing to find _timeSeriesDescriptor_.

Let me know what you find.

**Comment by ToddKopriva** [comment]
fixed by #1655
